### PR TITLE
[ALLUXIO-3090] Integrate Huawei OBS as under storage.

### PR DESCRIPTION
**Pull request by uronce-cc**
https://alluxio.atlassian.net/browse/ALLUXIO-3090

This PR integrates [Huawei OBS](http://support.huaweicloud.com/en-us/productdesc-obs/en-us_topic_0045853681.html) as Alluxio's under storage.

The Spark word count job at Spark's example directory `examples/src/main/python/wordcount.py` is successfully run on data stored in OBS through reading from Alluxio.

### CODE REVIEW

**Comment by uronce-cc** [comment]
@Xenorith The [Huawei OBS Java SDK](http://developer.huaweicloud.com/en-us/sdk/java/obs) can only be downloaded from Huawei's customized maven repository, which needs auth information set in `~/.m2/settings.xml`, how can I add this setting on the Jenkins servers?

**Comment by Xenorith** [comment]
I'm not familiar with how to configure this jenkins build, maybe @aaudiber or @apc999 know more?

**Comment by aaudiber** [comment]
@uronce-cc What is required to get credentials to download the SDK?

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSInputStream.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.InputStream;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * A stream for reading a file from OBS. This input stream returns 0 when calling read with an empty
+ * buffer.
+ */
+@NotThreadSafe
+public class OBSInputStream extends MultiRangeObjectInputStream {
```
Comment (madanadit) [inline underfs/obs/src/main/java/alluxio/underfs/oss/OBSInputStream.java]: Does ObsClient support an abort operation?

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSUnderFileSystem.java [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+/**
+ * Aliyun OSS {@link UnderFileSystem} implementation.
+ */
+@ThreadSafe
+public class OBSUnderFileSystem extends ObjectUnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(OBSUnderFileSystem.class);
+
+  /** Suffix for an empty file to flag it as a directory. */
+  private static final String FOLDER_SUFFIX = "_$folder$";
```
Comment (madanadit) [inline underfs/obs/src/main/java/alluxio/underfs/oss/OBSUnderFileSystem.java]: why not use "/"  as the suffix?

Comment (uronce-cc) [reply underfs/obs/src/main/java/alluxio/underfs/oss/OBSUnderFileSystem.java]: just to keep consistent with other similar underfs.

Comment (madanadit) [reply underfs/obs/src/main/java/alluxio/underfs/oss/OBSUnderFileSystem.java]: we use / with S3A and Swift. I would prefer using / for OBS as well

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSInputStream.java (hunk 1 above)
Comment (uronce-cc) [reply underfs/obs/src/main/java/alluxio/underfs/oss/OBSInputStream.java]: it only has operation to abort multipart upload, the SDK doc is at http://esdk.huawei.com/docs/obs/java-en/javadoc/index.html.

File: underfs/obs/src/main/java/alluxio/underfs/oss/OBSUnderFileSystem.java (hunk 2 above)
Comment (uronce-cc) [reply underfs/obs/src/main/java/alluxio/underfs/oss/OBSUnderFileSystem.java]: done

**Comment by uronce-cc** [comment]
Closing this PR to move it to https://github.com/Alluxio/alluxio-extensions

**Pull request closed without merging**